pandas/tests/indexes/common.py

```diff
@@ -635,6 +635,16 @@ def test_delete_base(self):
                 # either depending on numpy version
                 result = idx.delete(len(idx))
 
+    def test_equals(self):
+
+        for name, idx in compat.iteritems(self.indices):
+            self.assertTrue(idx.equals(idx))
+            self.assertTrue(idx.equals(idx.copy()))
+            self.assertTrue(idx.equals(idx.astype(object)))
+            self.assertFalse(idx.equals(list(idx)))
```

Comment:
> as base Index has the logic to check input class, i understand it is a original intension.

that is true. But the reality is that all other subclasses except base index do not have this check. Restricting those would be a back incompat API change, while we also have the option to relax it for the base class (which is also an API change, but IMO less likely to break code)